### PR TITLE
New gaussian linear calibration

### PR DESCRIPTION
**Pull request by mbaudin47**
This updates the implementation of gaussian linear calibration, in order to use a more stable code based on Cholesky decomposition combined with linear least squares.
Fixes #1165

### CODE REVIEW

**Comment by mbaudin47** [comment]
> 
> 
> Maybe we can have a test per UC ?
> See for example chaos (1 script for Ishigami UC / 1 script for Sobol functions )

Ok : I'll do that.

File: python/test/t_GaussianLinearCalibration_chaboche.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    covarianceDimension = size * outputDimension
+    observationDimension = localErrorCovariance.getDimension()
+    globalErrorCovariance = ot.CovarianceMatrix(covarianceDimension)
+    for i in range(size):
+        for j in range(observationDimension):
+            for k in range(observationDimension):
+                globalErrorCovariance[i * observationDimension + j, i * observationDimension + k] = localErrorCovariance[j,k]
+    return globalErrorCovariance
+
+def testCalibrationtChaboche():
```
Comment (sofianehaddad) [inline python/test/t_GaussianLinearCalibration_chaboche.py]: do we really need to define this function ?

File: python/test/t_GaussianLinearCalibration_std.py [hunk 2]
```diff
@@ -2,47 +2,110 @@
...
+    size = matrix.getNbRows()
+    dimension = matrix.getNbColumns()
+    sample = ot.Sample(size,dimension)
+    for i in range(size):
+        for j in range(dimension):
+            sample[i,j] = matrix[i,j]
+    return sample
+
+def testCalibrationtExponential():
+    '''
```
Comment (sofianehaddad) [inline python/test/t_GaussianLinearCalibration_std.py]: Same remark here : no need to define a function

Comment (mbaudin47) [reply python/test/t_GaussianLinearCalibration_std.py]: OK.

File: python/test/t_GaussianLinearCalibration_chaboche.py (hunk 1 above)
Comment (mbaudin47) [reply python/test/t_GaussianLinearCalibration_chaboche.py]: OK.

**Comment by mbaudin47** [comment]
@regislebrun : may I kindly suggest you to review this improved PR ? :)

File: lib/src/Uncertainty/Bayesian/GaussianLinearCalibration.cxx [hunk 3]
```diff
@@ -113,33 +113,55 @@ GaussianLinearCalibration::GaussianLinearCalibration(const Sample & modelObserva
...
-  const Matrix MtinvR(M.transpose() * invR);
-  const Matrix K((invB + MtinvR * M).solveLinearSystem(MtinvR));
-  const Point thetaStar(getCandidate() - K * deltaY);
-  const SquareMatrix L((IB - K * M).getImplementation());
-  const CovarianceMatrix covarianceThetaStar((K * R * K.transpose() + L * B * L.transpose()).getImplementation());
+  // Compute the inverse of the Cholesky decomposition of R
+  const Normal error(Point(R.getDimension()), R);
+  const TriangularMatrix errorInverseCholesky(error.getInverseCholesky());
+  // Compute errorInverseCholesky*J, the second part of the extended design matrix
+  const Matrix invLRJ = errorInverseCholesky * gradientObservations_;
```
Comment (regislebrun) [inline lib/src/Uncertainty/Bayesian/GaussianLinearCalibration.cxx]: `const Matrix invLRJ(errorInverseCholesky * gradientObservations_);`

Comment (mbaudin47) [reply lib/src/Uncertainty/Bayesian/GaussianLinearCalibration.cxx]: Ok. Thanks !

**Comment by mbaudin47** [comment]
The Travis error is : "Could not resolve host: github.com".

**Pull request merged**